editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/globalconfiguration/beans/ArgumentStyleChildTableControl.java

```diff
@@ -0,0 +1,362 @@
+ * Copyright (c) 2017 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/ 
+package org.fusesource.ide.camel.editor.globalconfiguration.beans;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.swing.event.ChangeEvent;
+import javax.swing.event.ChangeListener;
+
+import org.eclipse.core.runtime.ListenerList;
+import org.eclipse.jface.viewers.CellEditor;
+import org.eclipse.jface.viewers.ILabelProviderListener;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.jface.viewers.ITableLabelProvider;
+import org.eclipse.jface.viewers.ITreeContentProvider;
+import org.eclipse.jface.viewers.TextCellEditor;
+import org.eclipse.jface.viewers.TreeViewer;
+import org.eclipse.jface.viewers.Viewer;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.TreeColumn;
+import org.fusesource.ide.camel.model.service.core.model.AbstractCamelModelElement;
+import org.fusesource.ide.camel.model.service.core.model.CamelBasicModelElement;
+import org.fusesource.ide.camel.model.service.core.model.CamelFile;
+import org.w3c.dom.Element;
+
+/**
+ * @author brianf
+ *
+ */
+public class ArgumentStyleChildTableControl extends Composite {
+
+    public static final String TYPE_COLUMN = "type"; //$NON-NLS-1$
+    public static final String VALUE_COLUMN = "value"; //$NON-NLS-1$
+    private static final String[] TREE_COLUMNS = new String[] {TYPE_COLUMN, VALUE_COLUMN };
+
+    private Button addButton;
+    private Button removeButton;
+    private boolean isReadOnly = false;
+    private String warningMsg = null;
+    private ListenerList<ChangeListener> changeListeners;
+    private TreeViewer propertyTreeTable;
+    private AbstractCamelModelElement inputElement;
+    private List<AbstractCamelModelElement> argumentList = new ArrayList<>();
+
+    public ArgumentStyleChildTableControl(Composite parent, int style) {
+        this(parent, style, false);
+    }
+
+    public ArgumentStyleChildTableControl(Composite parent, int style, boolean isReadOnly) {
+		super(parent, style);
+
+        this.isReadOnly = isReadOnly;
+        this.changeListeners = new ListenerList<>();
+
+        int additionalStyles = SWT.NONE;
+        if (isReadOnly) {
```

Comment:
What do you think of writing the else too? It seems more readable for me.
int additionalStyles;
if(isReadOnly){
additionalStyles = SWT.READ_ONLY;
} else {
additionalStyles = SWT.NONE;
}